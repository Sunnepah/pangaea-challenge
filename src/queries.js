import { gql } from '@apollo/client';

export const GET_PRODUCTS = gql`
  query GET_PRODUCTS {
    products {
      id
      title
      image_url
      price(currency: USD)
    }
  }
`;
