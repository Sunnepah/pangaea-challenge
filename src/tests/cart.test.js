import TestRenderer, { create } from 'react-test-renderer';
import Cart from '../components/cart';
import { GET_ALL_CURRENCY } from '../queries';
import { MockedProvider } from '@apollo/client/testing';

const currrencyMock = [
  {
    request: {
      query: GET_ALL_CURRENCY,
    },
    result: {
      data: {
        currency: ['USD', 'NGN', 'EUR'],
      },
    },
  },
];

const { act } = TestRenderer;

test('renders the cart component with currency options', async () => {
  let cartComponent;

  await act(async () => {
    cartComponent = create(
      <MockedProvider mocks={currrencyMock} addTypename={false}>
        <Cart currency={{ symbol: '$', code: 'USD' }} />
      </MockedProvider>
    );
    return new Promise((resolve) => setTimeout(resolve, 0));
  });

  expect(cartComponent.toJSON()).toMatchSnapshot();
});
