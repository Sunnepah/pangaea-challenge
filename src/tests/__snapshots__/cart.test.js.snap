// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the cart component with currency options 1`] = `
<div
  className="cart__container"
>
  <div
    className="cart__body"
  >
    <div
      className="cart__body--top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <p>
        YOUR CART
      </p>
    </div>
    <select
      onChange={[Function]}
      value="USD"
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="NGN"
      >
        NGN
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
    </select>
    <p
      className="empty-cart"
    >
       There are no items in your cart
    </p>
  </div>
</div>
`;
